src/main/webapp/app/entities/exercise/exercise.model.ts

```diff
@@ -61,8 +61,33 @@ export abstract class Exercise implements BaseEntity {
     public numberOfParticipationsWithRatedResult: number;
     public numberOfSuccessfulParticipations: number;
     public averagePoints: number;
+    public numberOfAssessments: number;
 
     protected constructor(type: ExerciseType) {
         this.type = type;
     }
 }
+
+export function getIcon(exerciseType: ExerciseType): string {
+    const icons = {
+        [ExerciseType.PROGRAMMING]: 'keyboard',
+        [ExerciseType.MODELING]: 'project-diagram',
+        [ExerciseType.QUIZ]: 'check-double',
+        [ExerciseType.TEXT]: 'font',
+        [ExerciseType.FILE_UPLOAD]: 'file-upload',
+    };
+
+    return icons[exerciseType];
```

Comment:
I really like this change to the exercise icons

Reply:
Would it be faster, if we define the icons only once in a global constant? and not every time this method gets invoked? (which is probably quite often when you use it from html)

Reply:
@krusche this kind of optimizations do not work in Javascript:

- code is first transcompiled from Typescript to Javascript
- then, it is compiled by babel in a crazy one-liner blob that is served to the client
- finally, the browser engine does a compilation Just-In-Time(JIT) which is black magic.

In the end, it is better to just write good code without polluting the global namespace (making easy for babel to use shorter names).

Anyhow, I benchmarked the difference and there is no difference inside the error margin: https://jsperf.com/const-inside-vs-outside-function/1

Both of the example do more than 770 MILLIONS operation per second on my PC, definitely not a bottleneck